pyproject.toml

```
@@ -126,6 +125,7 @@ dependencies = [
     "types-urllib3",
     "types-jinja2",
     "types-babel",
+    "aiohttp==3.9.0b1; python_version >= '3.12'", # pre-release supports py3.12
```

```suggestion
    "aiohttp==3.9.0b1; python_version>='3.12'", # pre-release supports py3.12
```
Just for sake of consistency :)

---

These are not covered by GitHub Action, so pinning the version has no effect, and is rather detrimental for maintainance. If we go for GitHub Action, this is better achieved with `pip --pre` and marking as experimental

---

@LecrisUT Can you rephrase a bit? What you mean by GH Action?

All this fixes is that currently, on Fedora-39, `hatch env create dev` fails without explicit require on pre-release aiohttp. How would `pip --pre` come into this?

---

I guess the question is where is it executed? In `tmt`/`testing-farm`? In that case the packages would already be pre-installed.

For manually creating the environment, then it depends on how we create it via `venv` or `hatch` which would internally call `pip`. I am not sure on `hatch` side how one would change the pip repository and other settings, but maybe there is a `PIP_PRE` or equivalent environment variable to use globally. Haven't found the appropriate settings for that yet.

---

But that is what's confusing me, why are there no GitHub Actions or equivalent to actually run these or test with different environments.

---

Packit is running tests on all supported operating systems which have different Python versions.

---

> Packit is running tests on all supported operating systems which have different Python versions.

Sure, but that's through Fedora/EPEL packages, not through the hatch-env. It's just confusing where hatch-env sits betwern all of these environments. Right now it seems only for development environment, but in this case how is the IDE and development environment compared to vanila `venv`s? Do you have some tips on how to setup an environment.

Tangentially related, I've added sphinx spellcheck/linkcheck/etc. to a different project, and I was thinking to add it here as well. I have two possible implementations we ith vanilla GH Actions and with nox. Would you prefer a hatch + GH Actions solution for this instead?

---

@LecrisUT It is executed on _my_ Fedora-39 laptop I use to develop tmt.

---

> it here as well. I have two possible implementations we ith vanilla GH Actions and with nox. Would you prefer a hatch + GH Actions solution for this instead?

I'd say pre-commit. 

Its always better to be stopped before pushing then waiting on CI just to see that you need to amend.

---

> I'd say pre-commit.
> 
> Its always better to be stopped before pushing then waiting on CI just to see that you need to amend.

I'm not sure about this. It requires to run the whole documentation builds which might be a bit heavy for a pre-commit. Other tools usually support single file and diffs, but for spellcheck/doctest/linkcheck that's not the case. Also stuff like linkcheck are external, so I think it's better to be in GH action

---

Thanks everyone for discussion, with aiohttp released this change is not necessary any more ;)